yarn-project/circuits.js/src/tests/factories.ts

```diff
@@ -1315,11 +1320,13 @@ export function makeUnconstrainedFunctionWithMembershipProof(seed = 0): Unconstr
   };
 }
 
-export function makeContractClassPublic(seed = 0): ContractClassPublic {
+export function makeContractClassPublic(seed = 0, publicDispatchFunction?: PublicFunction): ContractClassPublic {
```

Comment:
We need to derive a "real" public contract class now since we do validation on contract id and address